File: PatientManagerServices/Enums/ExaminationType.cs
namespace PatientManagerServices.Models.Enums;

/// <summary>
/// Represents different types of medical examinations.
/// </summary>
public enum ExaminationType
{
    /// <summary>
    /// General physical examination.
    /// </summary>
    GP,

    /// <summary>
    /// Blood test.
    /// </summary>
    KRV,

    /// <summary>
    /// X-ray scan.
    /// </summary>
    X_RAY,

    /// <summary>
    /// CT scan.
    /// </summary>
    CT,

    /// <summary>
    /// MRI scan.
    /// </summary>
    MR,

    /// <summary>
    /// Ultrasound.
    /// </summary>
    ULTRA,

    /// <summary>
    /// Electrocardiogram.
    /// </summary>
    EKG,

    /// <summary>
    /// Echocardiogram.
    /// </summary>
    ECHO,

    /// <summary>
    /// Eye examination.
    /// </summary>
    EYE,

    /// <summary>
    /// Dermatological examination.
    /// </summary>
    DERM,

    /// <summary>
    /// Dental examination.
    /// </summary>
    DENTA,

    /// <summary>
    /// Mammography.
    /// </summary>
    MAMMO,

    /// <summary>
    /// Neurological examination.
    /// </summary>
    NEURO
}

File: PatientManagerApp/ClientApp/src/enums/ExaminationType.ts
export enum ExaminationType {
  GP,
  KRV,
  X_RAY,
  CT,
  MR,
  ULTRA,
  EKG,
  ECHO,
  EYE,
  DERM,
  DENTA,
  MAMMO,
  NEURO
}

export const ExaminationTypeProps = Object.keys(ExaminationType)
  .filter(key => isNaN(Number(key))) // Filter out numeric keys
  .map(key => ({
    title: key,
    value: ExaminationType[key as keyof typeof ExaminationType]
  }));

export const ExaminationTypeText = (type: number): string => {
  switch (type) {
    case ExaminationType.GP:
      return "General physical examination";
    case ExaminationType.KRV:
      return "Blood test";
    case ExaminationType.X_RAY:
      return "X-ray scan";
    case ExaminationType.CT:
      return "CT scan";
    case ExaminationType.MR:
      return "MRI scan";
    case ExaminationType.ULTRA:
      return "Ultrasound";
    case ExaminationType.EKG:
      return "Electrocardiogram";
    case ExaminationType.ECHO:
      return "Echocardiogram";
    case ExaminationType.EYE:
      return "Eye examination";
    case ExaminationType.DERM:
      return "Dermatological examination";
    case ExaminationType.DENTA:
      return "Dental examination";
    case ExaminationType.MAMMO:
      return "Mammography";
    case ExaminationType.NEURO:
      return "Neurological examination";
    default:
      return "Unknown examination type";
  }
};
